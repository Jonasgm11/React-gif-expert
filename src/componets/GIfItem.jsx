import PropTypes from "prop-types";
import React from 'react';

export const GifItem = ({title, url,id}) => {

  
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
        <p>Hola Jonas</p>
    </div>
  )
}

GifItem.proptypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}


