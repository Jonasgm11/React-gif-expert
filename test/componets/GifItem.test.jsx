import React from 'react';
import { GifItem } from "../../src/componets/GIfItem";
import { render, screen} from "@testing-library/react"

   

describe ('Pruebas en GIfItem', () => {

    const title = "Saitama"
    const url = "https://one-punch.com/saitama.jpg"

    test('DEBE DE HACER MATCH CON EL SNAPSHOT', () => {

       const {container} = 
       render (<GifItem title={title} url={url} />)
        expect (container).toMatchSnapshot();

    })
    
    test('DEBE MOSTRAR IMAGEN CON URL Y EL ATL INDICADO', () => {
      
        render (<GifItem title={title} url={url} />)
        // screen.debug();
        // expect (screen.getByRole('img').src).toBe(url)
        // expect (screen.getByRole('img').alt).toBe(title)

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })
    
        test('DEBE MOSTRAR EL TITULO EN EL COMPONENTE', () => {

            render (<GifItem title={title} url={url} />)
            expect (screen.getByAltText(title)).toBeTruthy();
        })
        

})