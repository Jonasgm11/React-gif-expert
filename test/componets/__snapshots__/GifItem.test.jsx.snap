// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GIfItem DEBE DE HACER MATCH CON EL SNAPSHOT 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
    <p>
      Hola Jonas
    </p>
  </div>
</div>
`;
